document pinning nixpkgs pin for NixOS by npins from `system.nix` (#1232)
nix-dot-dev/nix.dev: source/guides/recipes/dependency-management.md

## Managing NixOS configurations

NixOS defaults to using channels to locate `nixpkgs`.

> You can instead pin a version using `npins` from the `system.nix` entrypoint:

Reviewer: That makes it sound like "the `system.nix` entry point" is some sort of first-class concept.

Author: its [release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2605.section.md) entry mentions one could straight-up have their entire config in there, i.e. have a working system for the purpose of those `nixos-rebuild` / `nixos-install` without use of any other files.
that makes me wonder: what do you mean here when you'd consider it not first-class?

Reviewer: I wasn't aware of that change, but now that you remind me, I remember seeing some discussion around it when it was still in the works.

In that case, how about linking to the NixOS (or more precisely, `nixos-rebuild` and `nixos-install`) reference documentation on the subject? Otherwise the initial impression of "you have to know to know" still holds.

Author: as per the release notes, its use-case that needs passing some extra thing is when it's used as an attrset of nixos configurations, in which case the commands need to know which one to use.
that feature isn't invoked in the example here, and i've been able to use the `system.nix` entrypoint myself without really having needed to be aware of how it can be used: as far as on-par functionality goes, it 'just works'.

## Managing NixOS configurations

NixOS defaults to using channels to locate `nixpkgs`.
You can instead pin a version using `npins` from the `system.nix` entrypoint (available since 26.05):

```nix
let
  sources = import ./npins;
in
import "${sources.nixpkgs}/nixos" {
  configuration = ./configuration.nix;
}
```

> Before NixOS 26.05's `system.nix`, one may explicitly specify the `nixpkgs` pin like:

Reviewer 2: ```suggestion
Before NixOS 26.05's `system.nix` use:
```

I'll unify the nix.dev stylguide with the nixpkgs one.
But essentially: "Address the user as 'you'" and "Make the subject - In this case the user - do the action"

```bash
sudo NIX_PATH="nixos-config=configuration.nix:nixpkgs=$(nix-instantiate --raw --eval npins -A nixpkgs.outPath)" nixos-rebuild switch
```

> When using npins this way, you can disable channels in your configuration:

Reviewer 2: ```suggestion
If you use npins with system.nix, disable channels in your configuration:
```
